pkg/services/authn/clients/ext_jwt.go

```diff
@@ -34,28 +34,39 @@ const (
 
 func ProvideExtendedJWT(userService user.Service, cfg *setting.Cfg,
 	signingKeys signingkeys.Service) *ExtendedJWT {
-	verifier := authlib.NewVerifier[ExtendedJWTClaims](authlib.IDVerifierConfig{
+	verifier := authlib.NewVerifier[ExtendedJWTClaims](authlib.VerifierConfig{
 		SigningKeysURL: cfg.ExtJWTAuth.JWKSUrl,
 		AllowedAudiences: []string{
 			cfg.ExtJWTAuth.ExpectAudience,
 		},
-	})
+	}, authlib.TokenTypeAccess)
+
+	// The generic parameter is likely incorrect for ID Tokens
+	idTokenVerifier := authlib.NewVerifier[ExtendedJWTClaims](authlib.VerifierConfig{
+		SigningKeysURL: cfg.ExtJWTAuth.JWKSUrl,
+		AllowedAudiences: []string{
+			cfg.ExtJWTAuth.ExpectAudience,
+		},
+	}, authlib.TokenTypeID)
```

Comment:
We should probably used the typed version for these. So the first one can be replaced by https://github.com/grafana/authlib/blob/main/authn/verifier_access_token.go and the second one by https://github.com/grafana/authlib/blob/main/authn/verifier_id_token.go.

It not a blocker for this pr. I can fix that in another pr :)

Reply:
I addressed it, I think. The only difference from your recommendation is that I used the interface with generics over concrete types to allow for mocking in tests.

Reply:
Yep that is how we want to do it :100: